Adding Http Client

bobo333 opened the pull request:
I have a need to use HTTP instead of UDP (or TCP) for emitting metrics to Telegraf. This introduces an HTTP client to communicate with Telegraf. It issues the HTTP requests in a separate thread to avoid blocking the main thread.

I tested in both Python 2.7 and Python 3.5.

Commit:
Adding Http Client

bobo333 committed:
Adding requests-futures to dev-requirements.txt

bobo333 commented:
@paksu it appears the build in `python3.2` is failing because the `requests` library (and therefore the `requests-futures` library) doesn't support `python3.2`. The rationale [here](https://github.com/kennethreitz/requests/issues/3479) and [here](https://github.com/kennethreitz/requests/issues/3507) seems to indicate that it was a tradeoff to support unicode (with the `u'string'` syntax). A fix would be to no longer have this package support `python3.2` but I didn't want to make that change without your explicit approval.

bobo333 committed:
Remove Python3.2 Support

paksu commented:
Hey @bobo333 nice work. I'll try to review this soon.

paksu commented:
Hey @bobo333 your code looks good to me. The only thing I am hesitating about is the fact that that this adds a dependency that is not always needed. 

But this is a good addition so we should make it work for the users that want to use HTTP transport so I suggest you should make the following changes:

## Use `extra_requires` instead of `install_requires`
See the documentation here:
http://setuptools.readthedocs.io/en/latest/setuptools.html#declaring-extras-optional-features-with-their-own-dependencies

In the end setup.py should look like this:
```
setup(
   name='pytelegraf',
    ...
    extras_require={
        'http': ["requests-futures==0.9.7"],
    },
)
```

###  `HTTPClient` should import requests_futures
Because requests is not always installed we should only install it on demand. Maybe even throw a helpful error (i.e. `You need to install pytelegraf[http] if you need HTTP transport`)  if the import fails for some reason.

### Update the documentation about HTTPCLient
If the `requests-futures` is declared in the `extra_require` section then the documentation should mention that if the users want to use HTTP transport they should install pytelegraf like this:
```
pip install pytelegraf[http]==0.2.0
```
or add this to their requirements.txt

bobo333 committed:
Make HttpClient Dependencies Optional

Requires specifying http during install to install HttpClient's dependencies.
Only imports HttpClient's dependencies if using HttpClient. This allows for the use of the standard TelegrafClient without having to install requests-futures.

bobo333 commented:
@paksu thank you for reviewing! The feedback seemed reasonable to me, so I pushed up some changes that should address the comments.

paksu commented:
@bobo333 merged and uploaded the new version to pypi. Thanks again!

bobo333 commented:
Awesome, thanks for the quick turnaround on this @paksu
